SU2_CFD/src/solvers/CEulerSolver.cpp

```diff
@@ -1073,6 +1074,19 @@ void CEulerSolver::SetNondimensionalization(CConfig *config, unsigned short iMes
   }
   config->SetReThetaT_FreeStream(Re_ThetaT_FreeStream);
 
+  if (NcritNeeded) {
+    if (tkeNeeded) {
+      N_crit = -8.43 - 2.4 * log( (2.5 * tanh( config->GetTurbulenceIntensity_FreeStream() *100 / 2.5)) / 100.0);
+    }
+    else {
+      N_crit = 9.0;
+    }
+  } 
+  else {
+    N_crit = config->GetN_Critical();
+  }
+  config->SetN_Crtical(N_crit);
```

Comment:
Can this be done somewhere that lets be applicable to all flow solvers? Incompressible, NEMO, etc.